salt/modules/virt.py

```diff
@@ -2801,14 +2921,26 @@ def purge(vm_, dirs=False, removables=None, **kwargs):
     for disk in disks:
         if not removables and disks[disk]['type'] in ['cdrom', 'floppy']:
             continue
-        os.remove(disks[disk]['file'])
-        directories.add(os.path.dirname(disks[disk]['file']))
+        elif disks[disk].get('zfs', False):
+            # Sleep a little while to avoid 'dataset is busy'
+            time.sleep(3)
```

Comment:
That is fairly dangerous and can strike back. Is there no reliable way to know when the disk can be safely removed?

Reply:
After destroying a domain the OS seems to keep the volume openened for some time. I'm not sure yet how to create a better workaround for this other than waiting a bit.

Reply:
OK, then let's do it this way until you have a better idea